src/callbacks_step/amr_dg2d.jl

```diff
@@ -1,3 +1,95 @@
+# Redistribute data for load balancing after partitioning the mesh
+function redistribute_data!(u_ode::AbstractVector, mesh::TreeMesh{2}, equations, 
+                            dg::DGSEM, cache, old_mpi_ranks_per_cell)
+  # Retain current solution data
+  old_n_elements = nelements(dg, cache)
+  old_cell_ids = copy(cache.elements.cell_ids)
+  old_u_ode = copy(u_ode)
+  GC.@preserve old_u_ode begin # OBS! If we don't GC.@preserve old_u_ode, it might be GC'ed
+    old_u = wrap_array(old_u_ode, mesh, equations, dg, cache)
+
+    reinitialize_containers!(mesh, equations, dg, cache)
```

Comment:
```suggestion
    @timeit_debug timer() "reinitialize data structures" reinitialize_containers!(mesh, equations, dg, cache)
```

Reply:
This function is also called in `refine!` and `coarsen!`. Would you add a timer there too?

Reply:
No, that's not necessary I think. So far, this part of AMR was never a (big) issue in terms of performance, so I we can leave it for the time that someone needs to optimize the AMR routines.